src/sdk/analytics/index.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react'
+import { useAnalyticsEvent } from '@vtex/store-sdk'
+import type { PropsWithChildren } from 'react'
+
+export const AnalyticsHandler = ({ children }: PropsWithChildren<unknown>) => {
+  useEffect(() => {
+    window.dataLayer = window.dataLayer || []
```

Comment:
This makes a race condition possible if an analytics event is run before this effect. To elimitate this problem please move the dataLayer creation to the global scope. Also, make sure to make it work durring SSR
```suggestion
if (typeof window !== 'undefined') {
   window.dataLayer = window.dataLayer || []
}

export const AnalyticsHandler = ({ children }: PropsWithChildren<unknown>) => {
  useAnalyticsEvent((event) => {
    window.dataLayer.push(event)
  })
```